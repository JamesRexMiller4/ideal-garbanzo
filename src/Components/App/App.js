import React, { useState, useEffect } from 'react';
import './App.scss';
import Header from '../Header/Header';
import Form from '../Form/Form';
import ResultsContainer from '../ResultsContainer/ResultsContainer';
import Footer from '../Footer/Footer';
import { getData } from '../../apiCalls/apiCalls.js';
import * as utilFunctions from '../../utils/utilFunctions.js';

const App = () => {
  const [ data, setData ] = useState([]);
  const [ error, setError ] = useState('');  
  const [ results, setResults ] = useState([]);
  const [ page, setPage ] = useState(1);
  
  useEffect(() => {
    getData().then(data => {
      let cleanedData = data.map(restaurant => {
        return {
          id: restaurant["id"],
          name: restaurant["name"],
          address1: restaurant["address1"],
          city: restaurant["city"],
          state: restaurant["state"],
          zip: restaurant["zip"],
          telephone: restaurant["telephone"],
          tags: restaurant["tags"],
          website: restaurant["website"],
          genre: restaurant["genre"],
          hours: restaurant["hours"],
          attire: restaurant["attire"]
        }
      });
      
      setData(cleanedData)
      setResults(utilFunctions.alphabatizeResults(data))
    })
    .catch(error => setError(error))
  }, []);

  const resetResults = (setQuery, setSelectedState, setCheckedBoxes) => {
    setResults(utilFunctions.alphabatizeResults(data));
    setQuery('');
    setSelectedState('');
    setCheckedBoxes([]);
  };

  const setFilteredResults = (resultsData, query, selectedState, states, checkedBoxes) => {
    if (selectedState === "" && query === "" && checkedBoxes.length === 0) setResults(utilFunctions.alphabatizeResults(resultsData));

    let filteredResults = utilFunctions.filterByState(resultsData, selectedState, states);
    filteredResults = utilFunctions.filterByQuery(filteredResults, query);
    filteredResults = utilFunctions.filterByCheckboxes(filteredResults, checkedBoxes);
    
    setResults(filteredResults);
  };

  return (
    <div className="App">
      <Header />
      <Form 
        data={data}
        resetResults={resetResults}
        setFilteredResults={setFilteredResults}/>
      { data ? <ResultsContainer page={page} setPage={setPage} results={results}/> 
        : <h2>{error}</h2> }
      <Footer />
    </div>
  );
}

export default App;
