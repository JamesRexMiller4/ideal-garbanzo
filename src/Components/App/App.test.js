import React from 'react';
import { mount } from 'enzyme';
import App from './App';


describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App />)
  })
  it('should render the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
});