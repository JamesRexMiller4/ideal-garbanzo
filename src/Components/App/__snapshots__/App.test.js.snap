// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <Header>
      <header>
        <div
          className="logo-div"
        >
          <h1>
            Spectrum
          </h1>
          <img
            alt="play button"
            className="spectrum-icon"
            src="play.svg"
          />
        </div>
        <h2
          className="app-title"
        >
          Ideal Garbanzo Service
        </h2>
      </header>
    </Header>
    <Form
      data={Array []}
      resetResults={[Function]}
      setFilteredResults={[Function]}
    >
      <form
        className="main-form"
        onSubmit={[Function]}
      >
        <section
          className="regular-search-section"
        >
          <div
            className="select-state-div"
          >
            <label
              htmlFor="select-state"
            >
              Filter by State
            </label>
            <select
              id="select-state"
              name="select"
              onChange={[Function]}
              value=""
            >
              <option
                id=""
                key=""
                value=""
              />
              <option
                id="Alabama"
                key="Alabama"
                value="Alabama"
              >
                Alabama
              </option>
              <option
                id="Alaska"
                key="Alaska"
                value="Alaska"
              >
                Alaska
              </option>
              <option
                id="Arizona"
                key="Arizona"
                value="Arizona"
              >
                Arizona
              </option>
              <option
                id="Arkansas"
                key="Arkansas"
                value="Arkansas"
              >
                Arkansas
              </option>
              <option
                id="California"
                key="California"
                value="California"
              >
                California
              </option>
              <option
                id="Colorado"
                key="Colorado"
                value="Colorado"
              >
                Colorado
              </option>
              <option
                id="Connecticut"
                key="Connecticut"
                value="Connecticut"
              >
                Connecticut
              </option>
              <option
                id="Delaware"
                key="Delaware"
                value="Delaware"
              >
                Delaware
              </option>
              <option
                id="Florida"
                key="Florida"
                value="Florida"
              >
                Florida
              </option>
              <option
                id="Georgia"
                key="Georgia"
                value="Georgia"
              >
                Georgia
              </option>
              <option
                id="Hawaii"
                key="Hawaii"
                value="Hawaii"
              >
                Hawaii
              </option>
              <option
                id="Idaho"
                key="Idaho"
                value="Idaho"
              >
                Idaho
              </option>
              <option
                id="Illinois"
                key="Illinois"
                value="Illinois"
              >
                Illinois
              </option>
              <option
                id="Indiana"
                key="Indiana"
                value="Indiana"
              >
                Indiana
              </option>
              <option
                id="Iowa"
                key="Iowa"
                value="Iowa"
              >
                Iowa
              </option>
              <option
                id="Kansas"
                key="Kansas"
                value="Kansas"
              >
                Kansas
              </option>
              <option
                id="Kentucky"
                key="Kentucky"
                value="Kentucky"
              >
                Kentucky
              </option>
              <option
                id="Louisiana"
                key="Louisiana"
                value="Louisiana"
              >
                Louisiana
              </option>
              <option
                id="Maine"
                key="Maine"
                value="Maine"
              >
                Maine
              </option>
              <option
                id="Maryland"
                key="Maryland"
                value="Maryland"
              >
                Maryland
              </option>
              <option
                id="Massachusetts"
                key="Massachusetts"
                value="Massachusetts"
              >
                Massachusetts
              </option>
              <option
                id="Michigan"
                key="Michigan"
                value="Michigan"
              >
                Michigan
              </option>
              <option
                id="Minnesota"
                key="Minnesota"
                value="Minnesota"
              >
                Minnesota
              </option>
              <option
                id="Mississippi"
                key="Mississippi"
                value="Mississippi"
              >
                Mississippi
              </option>
              <option
                id="Missouri"
                key="Missouri"
                value="Missouri"
              >
                Missouri
              </option>
              <option
                id="Montana"
                key="Montana"
                value="Montana"
              >
                Montana
              </option>
              <option
                id="Nebraska"
                key="Nebraska"
                value="Nebraska"
              >
                Nebraska
              </option>
              <option
                id="Nevada"
                key="Nevada"
                value="Nevada"
              >
                Nevada
              </option>
              <option
                id="New Hampshire"
                key="New Hampshire"
                value="New Hampshire"
              >
                New Hampshire
              </option>
              <option
                id="New Jersey"
                key="New Jersey"
                value="New Jersey"
              >
                New Jersey
              </option>
              <option
                id="New Mexico"
                key="New Mexico"
                value="New Mexico"
              >
                New Mexico
              </option>
              <option
                id="New York"
                key="New York"
                value="New York"
              >
                New York
              </option>
              <option
                id="North Carolina"
                key="North Carolina"
                value="North Carolina"
              >
                North Carolina
              </option>
              <option
                id="North Dakota"
                key="North Dakota"
                value="North Dakota"
              >
                North Dakota
              </option>
              <option
                id="Ohio"
                key="Ohio"
                value="Ohio"
              >
                Ohio
              </option>
              <option
                id="Oklahoma"
                key="Oklahoma"
                value="Oklahoma"
              >
                Oklahoma
              </option>
              <option
                id="Oregon"
                key="Oregon"
                value="Oregon"
              >
                Oregon
              </option>
              <option
                id="Pennsylvania"
                key="Pennsylvania"
                value="Pennsylvania"
              >
                Pennsylvania
              </option>
              <option
                id="Rhode Island"
                key="Rhode Island"
                value="Rhode Island"
              >
                Rhode Island
              </option>
              <option
                id="South Carolina"
                key="South Carolina"
                value="South Carolina"
              >
                South Carolina
              </option>
              <option
                id="South Dakota"
                key="South Dakota"
                value="South Dakota"
              >
                South Dakota
              </option>
              <option
                id="Tennessee"
                key="Tennessee"
                value="Tennessee"
              >
                Tennessee
              </option>
              <option
                id="Texas"
                key="Texas"
                value="Texas"
              >
                Texas
              </option>
              <option
                id="Utah"
                key="Utah"
                value="Utah"
              >
                Utah
              </option>
              <option
                id="Vermont"
                key="Vermont"
                value="Vermont"
              >
                Vermont
              </option>
              <option
                id="Virginia"
                key="Virginia"
                value="Virginia"
              >
                Virginia
              </option>
              <option
                id="Washington"
                key="Washington"
                value="Washington"
              >
                Washington
              </option>
              <option
                id="West Virginia"
                key="West Virginia"
                value="West Virginia"
              >
                West Virginia
              </option>
              <option
                id="Wisconsin"
                key="Wisconsin"
                value="Wisconsin"
              >
                Wisconsin
              </option>
              <option
                id="Wyoming"
                key="Wyoming"
                value="Wyoming"
              >
                Wyoming
              </option>
            </select>
          </div>
          <div
            className="search-bar-div"
          >
            <input
              id="search-bar"
              onChange={[Function]}
              placeholder="Search Bar"
              type="text"
              value=""
            />
            <img
              alt="clear"
              className="clear-icon"
              onClick={[Function]}
              src="close.svg"
            />
            <img
              alt="search"
              className="search-icon"
              onClick={[Function]}
              src="search.svg"
            />
          </div>
          <div
            className="advanced-search-div"
          >
            <h3>
              Advanced Search
            </h3>
            <img
              alt="expand"
              id="advanced-search-btn"
              onClick={[Function]}
              src="downArrow.svg"
            />
          </div>
        </section>
        <section
          className="advanced-search-section"
        >
          <div
            className="filters-fieldsets-div"
          />
        </section>
      </form>
    </Form>
    <ResultsContainer
      results={Array []}
      setState={[Function]}
      state={
        Object {
          "data": Array [],
          "error": "",
          "page": 1,
          "results": Array [],
        }
      }
    >
      <section
        className="results-container-section"
      >
        <div
          className="results-pagination-div"
        >
          <img
            alt="previous-results"
            id="left-arrow"
            onClick={[Function]}
            src="leftArrow.svg"
          />
          <h2>
            Showing 1 of 1
          </h2>
          <img
            alt="more-results"
            id="right-arrow"
            onClick={[Function]}
            src="next.svg"
          />
        </div>
        <h2
          className="error"
        >
          No results, please try again
        </h2>
      </section>
    </ResultsContainer>
    <Footer>
      <footer>
        <div
          className="logo-div"
        >
          <h1>
            Spectrum
          </h1>
          <img
            alt="play"
            className="spectrum-icon"
            src="play.svg"
          />
        </div>
        <img
          alt="facebook"
          className="social-icon"
          src="facebook.svg"
        />
        <img
          alt="twitter"
          className="social-icon"
          src="twitter.svg"
        />
        <img
          alt="instagram"
          className="social-icon"
          src="instagram.svg"
        />
        <img
          alt="youtube"
          className="social-icon"
          src="youtube.svg"
        />
      </footer>
    </Footer>
  </div>
</App>
`;
